libarchive/archive_read_extract2.c

```diff
@@ -59,6 +67,9 @@ __archive_read_get_extract(struct archive_read *a)
 		}
 		a->cleanup_archive_extract = archive_read_extract_cleanup;
 	}
+  # ifndef NO_SYNC
+  sync();
```

Comment:
I don't understand this, either.  This function only returns an in-memory object.  There is no disk writing here.